matrix_generation/slurm.sh

```diff
@@ -1,12 +1,13 @@
 #!/bin/bash
 #SBATCH --job-name=Hamiltonian_Solver_agrace
-#SBATCH --mail-type=END,FAIL         # Mail events (NONE, BEGIN, END, FAIL, ALL)
+#SBATCH --mail-type=END        # Mail events (NONE, BEGIN, END, FAIL, ALL)
 #SBATCH --mail-user=agrace2@binghamton.edu
-#SBATCH --ntasks=18                   # Number of MPI tasks (i.e. processes)
+#SBATCH --ntasks=16                   # Number of MPI tasks (i.e. processes)
 #SBATCH --cpus-per-task=1            # Number of cores per MPI task 
-#SBATCH --nodes=6                    # Maximum number of nodes to be allocated
-#SBATCH --ntasks-per-node=6          # Maximum number of tasks on each node
-#SBATCH --ntasks-per-socket=6        # Maximum number of tasks on each socket
+#SBATCH --nodes=8                    # Maximum number of nodes to be allocated
+#SBATCH --acctg-freq=task=2
+#SBATCH --ntasks-per-node=4         # Maximum number of tasks on each node
+#SBATCH --ntasks-per-socket=4        # Maximum number of tasks on each socket
```

Comment:
@andygrace227 So I'm a little confused about this. Ideally, we want to minimize the number of processes. Inter-process communication is a killer. So ideally we want 1 or 2 (depending on the number of sockets and affinity) processes per node and do multi-threading parallelism in each node. Using message passing in a single node is not advisable unless absolutely necessary. 

Is it possible to run SLEPc in such a way, where it uses multi-threading and MPI?